classes/Component.spec.js

```diff
@@ -1,9 +1,269 @@
+        const template = component.buildTemplate();
+
+        const expectedImports = [templates.imports.react, templates.imports.connect];
+        const expectedBody = [templates.main].join('\n');
+        const expectedExport = [templates.exported.withConnect];
+        const expectedTemplate = expectedImports.join('\n') + '\n' + expectedBody + '\n' + expectedExport;
+
+        expect(template).to.equal(expectedTemplate);
+      });
+
+      it('style', () => {
+        const component = new Component('ComponentName', { style: true });
+        const template = component.buildTemplate();
+
+        const expectedImports = [templates.imports.react, '\n' + templates.imports.stylesheet];
+        const expectedBody = [templates.main].join('\n');
+        const expectedExport = [templates.exported.default];
+        const expectedTemplate = expectedImports.join('\n') + '\n' + expectedBody + '\n' + expectedExport;
+
+        expect(template).to.equal(expectedTemplate);
+      });
+
+      it('withConnect and style', () => {
+        const component = new Component(componentName, { withConnect: true, style: true });
+        const template = component.buildTemplate();
+
+        const expectedImports = [
+          templates.imports.react,
+          templates.imports.connect,
+          '\n' + templates.imports.stylesheet,
+        ];
+        const expectedBody = [templates.main].join('\n');
+        const expectedExport = [templates.exported.withConnect];
+        const expectedTemplate = expectedImports.join('\n') + '\n' + expectedBody + '\n' + expectedExport;
+
+        expect(template).to.equal(expectedTemplate);
+      });
+    });
+  });
+
+  describe('[writeComponentStructure]', () => {
+    let component;
+    let writeComponentFileStub;
+    let writeComponentIndexFileStub;
+    let manageComponentsIndexFileStub;
+
+    beforeEach(() => {
+      component = new Component(componentName);
+      writeComponentFileStub = sinon.stub(component, 'writeComponentFile');
+      writeComponentIndexFileStub = sinon.stub(component, 'writeComponentIndexFile');
+      manageComponentsIndexFileStub = sinon.stub(component, 'manageComponentsIndexFile');
+    });
+    afterEach(() => {
+      writeComponentFileStub.restore();
+      writeComponentIndexFileStub.restore();
+      manageComponentsIndexFileStub.restore();
+    });
+
+    describe('should write component structure', () => {
+      it('with component file if doesn\'t exist', () => {
+        const template = 'sample template';
+
+        component.writeComponentStructure(template);
+
+        expect(writeComponentFileStub).to.have.been.calledWith(template);
+        expect(writeComponentIndexFileStub).to.have.been.called;
+        expect(manageComponentsIndexFileStub).to.have.been.called;
+      });
+
+      it('without component file if already exists', () => {
+        mock({
+          [`/${componentName}/${componentName}.js`]: Buffer.from([8, 6, 7, 5, 3, 0, 9]),
```

Comment:
why you use Buffer here and why this array of numbers?

Reply:
It's just a mock for a artificial file.